### PR TITLE
Backend updates for My Projects work

### PR DESCRIPTION
**Pull request by cpeel**
When working on the new My Projects updates, various backend changes were made to support the new code. I pulled these out into their own commit to reduce the review overhead.

The two big changes in this MR:
* New tabbed interface for the Prefs and PM page.
* Move PP/PPV logic into Project class and out of the project.php page.

### CODE REVIEW

File: pinc/Project.inc [hunk 1]
```diff
@@ -133,40 +137,85 @@ class Project
...
+            $PPVer = $this->ppverifier;
+        }
+        else if ( !empty($this->checkedoutby) &&
+            in_array(
+                $this->state,
+                [
+                    PROJ_POST_SECOND_CHECKED_OUT,
+                    PROJ_POST_COMPLETE,
+                    PROJ_SUBMIT_PG_POSTED,
+                ]
```
Comment (srjfoo) [inline pinc/Project.inc]: This says pretty much the same thing as `private function _get_PPer()`, but `_get_PPer()` assigns the acceptable states to `$checkedout_states`. Functionality is the same, and I realize that the code was just moved from one place to another, but should they follow the same pattern?

Comment (cpeel) [reply pinc/Project.inc]: Yes, it would be better to have them more closely match. Fixed.

**Comment by cpeel** [comment]
@srjfoo, can you give one final look over the following two pages for how the tabs look and work?
* [PM page](https://www.pgdp.org/~cpeel/c/tools/project_manager/projectmgr.php)
* [Prefs page](https://www.pgdp.org/~cpeel/c/userprefs.php)

**Comment by srjfoo** [comment]
> @srjfoo, can you give one final look over the following two pages for how the tabs look and work?
> 
>     * [PM page](https://www.pgdp.org/~cpeel/c/tools/project_manager/projectmgr.php)
> 
>     * [Prefs page](https://www.pgdp.org/~cpeel/c/userprefs.php)

They appear to look and work as the tabs for `My Projects` do, and I don't see a difference in how the old tabs worked for `Prefs` and the updated version works. :)

**Pull request merged**